helix-term/src/ui/completion.rs

```diff
@@ -224,6 +226,12 @@ impl Completion {
                     // always present here
                     let item = item.unwrap();
 
+                    let offset_encoding = editor
+                        .language_servers
+                        .get_by_id(item.language_server_id)
+                        .expect("language server disappeared between completion request and application")
```

Comment:
I've refactored it a little bit with a macro that exits this function early in https://github.com/helix-editor/helix/pull/2507/commits/fb4cea53d76c6007ef25d1f458e812259fd95a6e. Better would indeed be closing the completion menu, but it's not completely trivial, as we need access to the compositor (which we have not...), so because it's such an edge_case, I just added a `TODO`

Reply:
yeah that's fair enough for now I will likely do significant changes to the life cycle of the completion menu in the future anyway so as long as we are not panicking that's good enough for now :+1: